components/core/src/clp_s/JsonParser.cpp

```diff
@@ -557,6 +621,307 @@ int32_t JsonParser::add_metadata_field(std::string_view const field_name, NodeTy
+        }
+        clp::streaming_compression::zstd::Decompressor decompressor;
+        size_t curr_pos = 0;
+        size_t last_pos = 0;
+        decompressor.open(file_path);
+
+        auto deserializer_result{Deserializer<IrUnitHandler>::create(decompressor, IrUnitHandler{})
+        };
+        if (deserializer_result.has_error()) {
+            decompressor.close();
+            m_archive_writer->close();
+            return false;
+        }
+        auto& deserializer = deserializer_result.value();
+        auto& ir_unit_handler{deserializer.get_ir_unit_handler()};
+
+        int32_t log_event_idx_node_id{};
+        auto add_log_event_idx_node = [&]() {
+            if (m_record_log_order) {
+                log_event_idx_node_id
+                        = add_metadata_field(constants::cLogEventIdxName, NodeType::Integer);
+            }
+        };
+        add_log_event_idx_node();
+        while (true) {
+            auto const kv_log_event_result{deserializer.deserialize_next_ir_unit(decompressor)};
+
+            if (kv_log_event_result.has_error()) {
+                break;
+            }
+            if (kv_log_event_result.value() == clp::ffi::ir_stream::IrUnitType::EndOfStream) {
+                break;
+            }
+            if (kv_log_event_result.value() == clp::ffi::ir_stream::IrUnitType::LogEvent) {
+                auto const kv_log_event = &(ir_unit_handler.get_deserialized_log_event().value());
+
+                m_current_schema.clear();
+
+                // Add log_event_idx field to metadata for record
+                if (m_record_log_order) {
+                    m_current_parsed_message.add_value(
+                            log_event_idx_node_id,
+                            m_archive_writer->get_next_log_event_id()
+                    );
+                    m_current_schema.insert_ordered(log_event_idx_node_id);
+                }
+
+                try {
+                    parse_kv_log_event(*kv_log_event);
+                } catch (std::exception const& e) {
+                    SPDLOG_ERROR("Encountered error while parsing a kv log event - {}", e.what());
+                    decompressor.try_get_pos(curr_pos);
+                    m_archive_writer->increment_uncompressed_size(curr_pos - last_pos);
+                    decompressor.close();
+                    return false;
+                }
+
+                if (m_archive_writer->get_data_size() >= m_target_encoded_size) {
+                    m_ir_node_to_archive_node_id_mapping.clear();
+                    decompressor.try_get_pos(curr_pos);
+                    m_archive_writer->increment_uncompressed_size(curr_pos - last_pos);
+                    last_pos = curr_pos;
+                    split_archive();
+                    add_log_event_idx_node();
+                }
+
+                ir_unit_handler.clear();
+                m_current_parsed_message.clear();
+
+            } else {
+                continue;
```

Comment:
```suggestion
                m_archive_writer->close();
                decompressor.close();
                return false;
```
Sorry, missed this previously again, but we should error out on any IR unit we don't know how to handle. Want to avoid cases where we silently don't handle some feature and ingest an IR stream incorrectly.

For example, after reading through the IRv2 serializer/deserializer code a bit, it seems like we don't handle the UTC offset change feature. If we allow ingestion of streams that have changing UTC offsets then we'll create archives with incorrect timestamps.

Reply:
This case covers both SchemaTreeNodeInsertion and UtcOffsetChange IR unit types. The SchemaTreeNodeInsertion is perfectly fine. we just don't need to do anything on it. I'll special case that one and fail in the else case.